step.sh

```diff
@@ -1,27 +1,40 @@
 #!/bin/bash
 set -e
 
-echo "Checking status of collector at $collector_url..."
-result=$(curl --silent "$collector_url/micro/all")
+echo "Checking status of collector at ${CP_BITRISE_SNOWPLOW_MICRO_COLLECTOR_URL}/micro/all"
+result=$(curl --silent "${CP_BITRISE_SNOWPLOW_MICRO_COLLECTOR_URL}/micro/all")
+
+echo result
 
 # Parse the results
-total=$(jq '.total' <<< "$result")
-good=$(jq '.good' <<< "$result")
-bad=$(jq '.bad' <<< "$result")
+total=$(jq -r '.total' <<< "$result")
+good=$(jq -r '.good' <<< "$result")
+bad=$(jq -r '.bad' <<< "$result")
 
 # Export them to variables
 echo "Parsed results (total: $total, good: $good, bad: $bad)"
 envman add --key SNOWPLOW_MICRO_COLLECTOR_RESULTS_TOTAL --value "$total"
 envman add --key SNOWPLOW_MICRO_COLLECTOR_RESULTS_GOOD --value "$good"
 envman add --key SNOWPLOW_MICRO_COLLECTOR_RESULTS_BAD --value "$bad"
 
-# TODO: Do a lot more.. Export the bad results (log files? test results?)
-# TODO: Offer an option to reset?
-# TODO: Maybe use a better language, Go? or Ruby?
+# TODO: In the future
+# We need to do bad events parsing according to https://docs.snowplowanalytics.com/docs/managing-data-quality/testing-and-qa-workflows/set-up-automated-testing-with-snowplow-micro/#micro-bad
+# Then create a junit report file that lists each issue as a failure, write it to $BITRISE_TEST_RESULT_DIRCreate a junit report file that lists each issue as a failure, write it to $BITRISE_TEST_RESULT_DIR
+# It will be sharable script codes among other clients like `Android` and `Web`
 
 # Fail if bad is != 0
-if [ "$bad" != "0" ];
+if [[ $total == "0" ]]
+then
+  echo "There are no events posted to Snowplow Micro, text failed."
+  exit 1
+elif [[ $bad == "0" ]]
 then
+  echo "Snowplow test was successfully. :tada"
```

Comment:
I think you will need to use actual emoji here instead of `:tada:` 🎉 

```suggestion
  echo "Snowplow test was successful 🎉"
```